data/common/__init__.py

```diff
@@ -214,3 +215,25 @@ def fix_unit(unit):
             return "t⋅km"
         case _:
             return unit
+
+
+def format_number(value):
+    """Format a number to a float with 6 significant digits."""
+    if value is None:
+        return "None"
+    return float(f"{value:.6g}")
+
+
+class FormatNumberJsonEncoder(json.JSONEncoder):
+    def encode(self, obj):
+        def recursive_format_number(obj):
+            if isinstance(obj, (int, float)):
+                return format_number(obj)
```

Comment:
Do you really need a function call here as you don't use it anywhere else?
```suggestion
                return float(f"{obj:.6g}")
```